packages/shared/src/components/filters/TagOptionsMenu.tsx

```diff
@@ -29,30 +33,46 @@ export default function TagOptionsMenu({
   onFollow,
   onUnfollow,
 }: TagOptionsMenuProps): ReactElement {
+  const { isOpen } = useContextMenu({
+    id: ContextMenuIds.TagOptionsContext,
+  });
+  const tagOptions: MenuItemProps[] = [];
+
+  if (tag) {
+    tagOptions.push({
+      label: 'View',
+      anchorProps: {
+        href: getTagPageLink(tag.name ?? (tag as string)),
+      },
+    });
+  }
+
+  if (onFollow) {
+    tagOptions.push({ label: 'Follow', action: onFollow });
+  }
+
+  if (onUnfollow) {
+    tagOptions.push({ label: 'Unfollow', action: onUnfollow });
+  }
+
+  if (onBlock) {
+    tagOptions.push({ label: 'Block', action: onBlock });
+  }
+
+  if (onUnblock) {
+    tagOptions.push({ label: 'Unblock', action: onUnblock });
+  }
```

Comment:
We could extract this in a separate func, a helper or sth to gives the tag options

Reply:
Probably a useMemo can help?